catalyst_voices/lib/widgets/rich_text/voices_rich_text.dart

```diff
@@ -0,0 +1,256 @@
+import 'package:catalyst_voices/widgets/buttons/voices_filled_button.dart';
+import 'package:catalyst_voices/widgets/common/resizable_box_parent.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_quill/extensions.dart';
+import 'package:flutter_quill/flutter_quill.dart';
+import 'package:flutter_quill_extensions/flutter_quill_extensions.dart';
+
+/// A component for rich text writing
+/// using Quill under the hood
+/// https://pub.dev/packages/flutter_quill
+class VoicesRichText extends StatefulWidget {
+  const VoicesRichText({
+    super.key,
+    this.document,
+    this.onSave,
+    this.charsLimit,
+  });
+
+  final Document? document;
+  final ValueChanged<Document>? onSave;
+  final int? charsLimit;
+
+  @override
+  State<VoicesRichText> createState() => _VoicesRichTextState();
+}
+
+class _VoicesRichTextState extends State<VoicesRichText> {
+  final QuillController _controller = QuillController.basic();
+  int _documentLength = 0;
+  bool _editMode = false;
+  FocusNode _focusNode = FocusNode();
+
+  void _onDocumentChange(DocChange docChange) {
```

Comment:
consider to put it below initState(), would prefer that we order methods in a class according to due chronology, i.e. in order they are called by the framework, in order they call themselves, etc.